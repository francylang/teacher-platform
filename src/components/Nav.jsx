import React, { Component } from 'react';
import Search from './Search';

class Nav extends Component {
  render() {
    const { handleSearch, allDiscussions, discussions, renderDiscussions, renderFilteredDiscussions } = this.props;
    return (
      <div className="nav">
        <Search
          handleSearch={handleSearch}
          renderDiscussions={renderDiscussions}
          discussions={discussions}
          allDiscussions={allDiscussions}
          renderFilteredDiscussions={renderFilteredDiscussions}
        />
        <button
          onClick={() => this.props.renderStandards()}
          className="filter-btn"
        >
          <h4 className="standard-option">
            Standards
          </h4>
        </button>
        <button
          onClick={() => this.props.renderForm()}
          className="toggle-discussion-form-btn"
          value="form">
          <h4 className="new-discussion-option">
            New Discussion
          </h4>
        </button>
        <button
          onClick={() => this.props.renderDiscussions()}
          className="toggle-discussions-btn"
          value="form">
          <h4 className="new-discussion-option">
            All Discussions
          </h4>
        </button>
      </div>
    );
  }
}

export default Nav;
