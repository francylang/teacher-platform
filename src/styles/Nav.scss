.nav {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #EAF2F8;
  border: 1px solid #EAF2F8;
  padding: 0 2%;
}

.filter-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: #EAF2F8;
  border: none;
  outline: none;
  cursor: pointer;
}

.standard-option {
  padding-right: 10px;
  color: #2B2B2B;
}

.toggle-discussion-form-btn,
.toggle-discussions-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: #EAF2F8;
  border: none;
  outline: none;
  cursor: pointer;
  padding-right: 0;
}

.new-discussion-option {
  color: #2B2B2B;
}
