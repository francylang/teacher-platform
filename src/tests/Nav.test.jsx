import React from 'react';
import ReactDOM from 'react-dom';
import Nav from '../components/Nav';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<Nav />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Nav />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });
  
  it('should render a nav', () => {
    const nav = wrapper.find('.nav');

    expect(nav.length).toEqual(1);
  });

  it('should render 3 buttons', () => {
    const button = wrapper.find('button');

    expect(button.length).toEqual(3);
  });

  it('should render 3 h4s', () => {
    const h4 = wrapper.find('h4');

    expect(h4.length).toEqual(3);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
