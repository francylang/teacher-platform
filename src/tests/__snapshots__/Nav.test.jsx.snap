// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Search)
          filteredDiscussions={undefined}
          handleSearch={undefined}
          renderDiscussions={undefined}
          renderFilteredDiscussions={undefined}
/>,
        <button
          className="filter-btn"
          onClick={[Function]}
>
          <h4
                    className="standard-option"
          >
                    Standards
          </h4>
</button>,
        <button
          className="toggle-discussion-form-btn"
          onClick={[Function]}
          value="form"
>
          <h4
                    className="new-discussion-option"
          >
                    New Discussion
          </h4>
</button>,
        <button
          className="toggle-discussions-btn"
          onClick={[Function]}
          value="form"
>
          <h4
                    className="new-discussion-option"
          >
                    All Discussions
          </h4>
</button>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filteredDiscussions": undefined,
          "handleSearch": undefined,
          "renderDiscussions": undefined,
          "renderFilteredDiscussions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4
            className="standard-option"
>
            Standards
</h4>,
          "className": "filter-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Standards",
            "className": "standard-option",
          },
          "ref": null,
          "rendered": "Standards",
          "type": "h4",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4
            className="new-discussion-option"
>
            New Discussion
</h4>,
          "className": "toggle-discussion-form-btn",
          "onClick": [Function],
          "value": "form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Discussion",
            "className": "new-discussion-option",
          },
          "ref": null,
          "rendered": "New Discussion",
          "type": "h4",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4
            className="new-discussion-option"
>
            All Discussions
</h4>,
          "className": "toggle-discussions-btn",
          "onClick": [Function],
          "value": "form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Discussions",
            "className": "new-discussion-option",
          },
          "ref": null,
          "rendered": "All Discussions",
          "type": "h4",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Search)
            filteredDiscussions={undefined}
            handleSearch={undefined}
            renderDiscussions={undefined}
            renderFilteredDiscussions={undefined}
/>,
          <button
            className="filter-btn"
            onClick={[Function]}
>
            <h4
                        className="standard-option"
            >
                        Standards
            </h4>
</button>,
          <button
            className="toggle-discussion-form-btn"
            onClick={[Function]}
            value="form"
>
            <h4
                        className="new-discussion-option"
            >
                        New Discussion
            </h4>
</button>,
          <button
            className="toggle-discussions-btn"
            onClick={[Function]}
            value="form"
>
            <h4
                        className="new-discussion-option"
            >
                        All Discussions
            </h4>
</button>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filteredDiscussions": undefined,
            "handleSearch": undefined,
            "renderDiscussions": undefined,
            "renderFilteredDiscussions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="standard-option"
>
              Standards
</h4>,
            "className": "filter-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Standards",
              "className": "standard-option",
            },
            "ref": null,
            "rendered": "Standards",
            "type": "h4",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="new-discussion-option"
>
              New Discussion
</h4>,
            "className": "toggle-discussion-form-btn",
            "onClick": [Function],
            "value": "form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New Discussion",
              "className": "new-discussion-option",
            },
            "ref": null,
            "rendered": "New Discussion",
            "type": "h4",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="new-discussion-option"
>
              All Discussions
</h4>,
            "className": "toggle-discussions-btn",
            "onClick": [Function],
            "value": "form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All Discussions",
              "className": "new-discussion-option",
            },
            "ref": null,
            "rendered": "All Discussions",
            "type": "h4",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
